### PR TITLE
IG-16932

### PR DESCRIPTION
**Pull request by talIguaz**
* enable querying multiple targets in grafana
* fix inconsistent results due to labels order inconsistency (in `getBaseTargetTSDB`)

### CODE REVIEW

File: http/grafana.go [hunk 1]
```diff
@@ -100,21 +99,38 @@ type timeSeriesOutput struct {
...
+		return nil, err
+	}
+	var requests []simpleJSONRequestInterface
 	switch method {
 	case "query":
-		retval = &simpleJSONQueryRequest{Backend: defaultBackend}
+		for _, target := range reqBase.Targets {
+			currRequest := &simpleJSONQueryRequest{Backend: defaultBackend, requestSimpleJSONBase: reqBase}
+			currRequest.Type = target["type"].(string)
+			fieldInput := target["target"].(string)
```
Comment (gtopper) [inline http/grafana.go:113]: Doing it like that means that the service will panic, crash and restart on a bad input.

File: http/grafana.go [hunk 2]
```diff
@@ -100,21 +99,38 @@ type timeSeriesOutput struct {
...
+			if err := currRequest.parseQueryLine(fieldInput); err != nil {
+				return nil, err
```
Comment (gtopper) [inline http/grafana.go:115]: Should probably wrap this error somewhere, or the user will get some regex error out of context.

File: http/server.go [hunk 3]
```diff
@@ -449,14 +449,42 @@ func (s *Server) handleSimpleJSONSearch(ctx *fasthttp.RequestCtx) {
...
 		ctx.Error(fmt.Sprintf("bad request - %s", err), http.StatusBadRequest)
 		return
 	}
-	// passing session in order to easily pass token, when implemented
+
+	var results interface{}
+	for _, req := range requests {
+		result, err := s.executeSimpleJSONSubRequest(req, ctx)
+		if err != nil {
+			ctx.Error(fmt.Sprintf("Error querying: %s", err), http.StatusInternalServerError)
```
Comment (gtopper) [inline http/server.go:463]: `%v` or `err.Error()`

File: http/server.go [hunk 4]
```diff
@@ -449,14 +449,42 @@ func (s *Server) handleSimpleJSONSearch(ctx *fasthttp.RequestCtx) {
...
+
+func appendSimpleJSONResults(origin, resToAdd interface{}) interface{} {
+	if origin == nil {
+		return resToAdd
+	}
+	switch origin.(type) {
+	case []timeSeriesOutput:
+		return append(origin.([]timeSeriesOutput), resToAdd.([]timeSeriesOutput)...)
+	default:
+		return append(origin.([]tableOutput), resToAdd.([]tableOutput)...)
```
Comment (gtopper) [inline http/server.go:481]: Unsafe 💥🚗

File: http/server.go [hunk 5]
```diff
@@ -449,14 +449,42 @@ func (s *Server) handleSimpleJSONSearch(ctx *fasthttp.RequestCtx) {
...
+	}
+}
+
+func (s *Server) executeSimpleJSONSubRequest(req simpleJSONRequestInterface, ctx *fasthttp.RequestCtx) (interface{}, error) {
 	request := req.GetReadRequest(nil)
+	fmt.Println(request)
```
Comment (gtopper) [inline http/server.go:487]: ❌

File: http/server.go [hunk 6]
```diff
@@ -469,18 +497,16 @@ func (s *Server) handleSimpleJSON(ctx *fasthttp.RequestCtx, method string) {
...
-		if apiError != nil {
-			s.logger.ErrorWith("error reading", "error", apiError)
-		}
 	}()
-	if apiError != nil {
-		ctx.Error(fmt.Sprintf("Error creating response - %s", apiError), http.StatusInternalServerError)
+
+	resp, err := CreateResponse(req, ch)
+	if err != nil {
+		return nil, err
```
Comment (gtopper) [inline http/server.go:504]: Better wrap it? Or the client will just get a "Bad format" error.

File: http/server.go [hunk 7]
```diff
@@ -449,14 +449,42 @@ func (s *Server) handleSimpleJSONSearch(ctx *fasthttp.RequestCtx) {
...
+	}
+
+	_ = s.replyJSON(ctx, results)
+}
+
+func appendSimpleJSONResults(origin, resToAdd interface{}) interface{} {
+	if origin == nil {
+		return resToAdd
+	}
+	switch origin.(type) {
```
Comment (gtopper) [inline http/server.go:477]: `switch x := origin.(type) {`

File: http/server.go (hunk 4 above)
Comment (talIguaz) [reply http/server.go:481]: Safe enough for us, these types and all are internal.

**Review by gtopper — APPROVED** [review]

**Pull request merged**